ddsc/core/fileuploader.py

```diff
@@ -118,10 +124,24 @@ def send_file_external(self, url_json, chunk):
         host = url_json['host']
         url = url_json['url']
         http_headers = url_json['http_headers']
-        resp = self.data_service.send_external(http_verb, host, url, http_headers, chunk)
+        resp = self._send_file_external_with_retry(http_verb, host, url, http_headers, chunk)
         if resp.status_code != 200 and resp.status_code != 201:
             raise ValueError("Failed to send file to external store. Error:" + str(resp.status_code))
 
+    def _send_file_external_with_retry(self, http_verb, host, url, http_headers, chunk):
+        """
+        Send chunk to host, url using http_verb. If the http_verb is PUT and a connection error occurs
+        retry once after waiting.
+        """
+        try:
+            return self.data_service.send_external(http_verb, host, url, http_headers, chunk)
+        except requests.exceptions.ConnectionError:
```

Comment:
In testing last week one retry was enough but no idea how many we might need. I'll bump it up to 3 but that's also just a guess.